account-registration/src/main/java/com/amazon/aws/spinnaker/plugin/registration/AccountsStatus.java

```diff
@@ -135,14 +136,29 @@ public synchronized boolean getDesiredAccounts() {
             setBackoffTime();
             return false;
         }
+        
+        // Handle null or empty accounts lists
+        if (response.getAccounts() == null) {
+            log.warn("Response accounts list is null, treating as empty list");
+            response.setAccounts(new ArrayList<>());
```

Comment:
https://github.com/spinnaker-plugins/aws-account-registration-plugin-spinnaker/blob/1.3.0/account-registration/src/main/java/com/amazon/aws/spinnaker/plugin/registration/AccountsStatus.java#L233 I'd HAVE THOUGHT... this would have covered some of this... 
```
            response.setAccounts(Collections.emptyList());
```
Of course that sets the account to an immutable list which woudl break areas below.